aws/src/integration/README.md

```diff
@@ -0,0 +1,197 @@
+To add new permissions in CI, please contact [dev@iceberg.apache.org](mailto:dev@iceberg.apache.org).
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "S3BucketPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetAccessPoint",
+        "s3:CreateAccessPoint",
+        "s3:DeleteAccessPoint",
+        "s3:ListAccessPoints",
+        "s3:ListBucket"
+      ],
+      "Resource": [
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-east-1",
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-west-2"
+      ]
+    },
+    {
+      "Sid": "S3ObjectPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:GetObjectAcl",
+        "s3:GetObject",
+        "s3:DeleteObject",
+        "s3:AbortMultipartUpload"
+      ],
+      "Resource": [
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-east-1/*",
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-west-2/*"
+      ]
+    },
+    {
+      "Sid": "GluePermissions",
+      "Effect": "Allow",
+      "Action": [
+        "glue:UpdateDatabase",
+        "glue:DeleteDatabase",
+        "glue:CreateTable",
+        "glue:GetTables",
+        "glue:GetTableVersions",
+        "glue:UpdateTable",
+        "glue:DeleteTable",
+        "glue:GetDatabases",
+        "glue:GetTable",
+        "glue:GetDatabase",
+        "glue:GetTableVersion",
+        "glue:CreateDatabase"
+      ],
+      "Resource": [
+        "arn:aws:glue:us-east-1:220427471354:catalog",
+        "arn:aws:glue:us-east-1:220427471354:database/iceberg_aws_ci_*",
+        "arn:aws:glue:us-east-1:220427471354:table/iceberg_aws_ci_*/*",
+        "arn:aws:glue:us-east-1:220427471354:userDefinedFunction/iceberg_aws_ci_*/*"
+      ]
+    },
+    {
+      "Sid": "STSPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "sts:TagSession",
+        "sts:AssumeRole"
+      ],
+      "Resource": [
+        "arn:aws:iam::220427471354:role/iceberg-aws-ci-*"
+      ]
+    },
+    {
+      "Sid": "KMSKeyCreationPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "kms:CreateKey"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "KMSAliasPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "kms:CreateAlias"
+      ],
+      "Resource": [
+        "arn:aws:kms:us-east-1:220427471354:key/*",
+        "arn:aws:kms:us-east-1:220427471354:alias/iceberg-aws-ci-*"
+      ]
+    },
+    {
+      "Sid": "KMSKeyUsagePermissions",
+      "Effect": "Allow",
+      "Action": [
+        "kms:Decrypt",
+        "kms:Encrypt",
+        "kms:GenerateDataKey",
+        "kms:ScheduleKeyDeletion"
+      ],
+      "Resource": "arn:aws:kms:us-east-1:220427471354:key/*",
+      "Condition": {
+        "ForAnyValue:StringLike": {
+          "kms:ResourceAliases": "alias/iceberg-aws-ci-*"
+        }
+      }
+    },
+    {
+      "Sid": "IAMPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateRole",
```

Comment:
looking at the policy here, we allow create role, put role policy and then assume role. This is not good because if the credential is leaked the caller will be able to create any role with any policy and then assume the role, which basically grants admin access to the account. My current thought is that we should exclude AssumeRole related tests from integration test execution. Any thoughts? @amogh-jahagirdar @rajarshisarkar @singhpk234 @xiaoxuandev

Reply:
Actually, we can pre-create an IAM role with the same set of access policies, and the CI IAM user just have one more permission of assume the created role, without permission to touch any role policy. This also avoids the need to wait for IAM consistency, so it's likely a better way.

Reply:
#4909